crates/kornia-imgproc/src/features/fast.rs

```diff
@@ -133,6 +155,236 @@ fn is_fast_corner(
     false
 }
 
+/// Calculate the FAST corner score for a pixel using Sum of Absolute Differences (SAD).
+/// Based on https://www.edwardrosten.com/work/rosten_2006_machine.pdf.
+/// Returns a tuple of (is_corner: bool, score: i32)
+fn get_fast_corner_score(
+    src: &[u8],
+    pixel_idx: i32,
+    offsets: [i32; 16],
+    threshold: u8,
+    arc_length: u8,
+) -> (bool, i32) {
+    let center_pixel = unsafe { *src.get_unchecked(pixel_idx as usize) };
+    let lower_threshold = &center_pixel.saturating_sub(threshold);
+    let upper_threshold = &center_pixel.saturating_add(threshold);
+
+    // Helper to get pixel value efficiently with unchecked access
+    let get_pixel_from_offset =
+        |off_idx: usize| unsafe { src.get_unchecked((pixel_idx + offsets[off_idx]) as usize) };
+
+    // Fast rejection test - check if at least 3 of the 4 high-speed test points are different enough
+    let p1 = get_pixel_from_offset(0);
+    let p5 = get_pixel_from_offset(4);
+    let p9 = get_pixel_from_offset(8);
+    let p13 = get_pixel_from_offset(12);
+
+    let m0 = (p1 >= lower_threshold) as u8 + 
+                (p5 >= lower_threshold) as u8 + 
+                (p9 >= lower_threshold) as u8 + 
+                (p13 >= lower_threshold) as u8;
+
+    let m1 = (p1 <= upper_threshold) as u8 + 
+                (p5 <= upper_threshold) as u8 + 
+                (p9 <= upper_threshold) as u8 + 
+                (p13 <= upper_threshold) as u8;
+
+    if m0 < 3 && m1 < 3 {
+        return (false, 0);
+    }
+
+    // check the remaining pixels
+    let p2 = get_pixel_from_offset(1);
+    let p3 = get_pixel_from_offset(2);
+    let p4 = get_pixel_from_offset(3);
+    let p6 = get_pixel_from_offset(5);
+    let p7 = get_pixel_from_offset(6);
+    let p8 = get_pixel_from_offset(7);
+    let p10 = get_pixel_from_offset(9);
+    let p11 = get_pixel_from_offset(10);
+    let p12 = get_pixel_from_offset(11);
+    let p14 = get_pixel_from_offset(13);
+    let p15 = get_pixel_from_offset(14);
+    let p16 = get_pixel_from_offset(15);
+    let pixels = [
+        p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12, p13, p14, p15, p16,
+        p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12, p13, p14, p15, p16 
+    ]; // Values are repeated to handle circular indexing
+    
+    let mut consecutive_brighter = 0u8;
+    let mut consecutive_darker = 0u8;
+
+    let mut is_corner = false;
+    let mut start_idx = 0;
+    let mut end_idx = 0;
```

Comment:
This is not updated right ?

Reply:
All three are used to track the largest segment which is used calculate the NMS score for the center pixel.